File: Lab2/Lab2/Plane.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab2
{
    public abstract class Plane:ITransport
    {


        protected float startX;
        protected float startY;

		public virtual int maxSpeed { protected set; get; }
        public virtual int weight { protected set; get; }
		public virtual int maxHeight { protected set; get; }
        public virtual int fuel { protected set; get; }

        public Color colorBody { protected set; get; }

        public abstract void move(Graphics g);
        public abstract void draw(Graphics g);

        public void setPosition(int x,int y)
        {
            startX = x;
            startY = y;
        }
        
        public virtual void setMainColor(Color c)
		{
			colorBody = c;
		}


    }
}

File: Lab2/src/Plane.java
import java.awt.Color;
import java.awt.Graphics;

public abstract class Plane implements ITransport {

	protected int startX;
	protected int startY;
	public int maxSpeed;
	public int weight;
	public int maxHeight;
	public int fuel;
	public Color color;

	protected abstract void setmaxSpeed(int s);

	public abstract int getmaxSpeed();

	protected abstract void setWeight(int w);

	public abstract int getWeight();

	protected abstract void setmaxHeight(int h);

	public abstract int getmaxHeight();

	public abstract void move(Graphics g);

	public abstract void draw(Graphics g);

	public void setPosition(int x, int y) {
		startX = x;
		startY = y;
	}

	public void setMainColor(Color c) {
		color = c;
	}
}
